components/NameGeneratorForm/NameGeneratorForm.tsx

```diff
@@ -1,58 +1,49 @@
 import { FC } from "react"
+import { useNFTNameGenerator } from "../../providers/NFTNameGeneratorProvider"
 import MintButton from "../MintButton"
 
-interface NameGeneratorFormProps {
-  twitterHandle: string
-  setTwitterHandle: (value: string) => void
-  publicWallet: string
-  setPublicWallet: (value: string) => void
-  loading: boolean
+const NameGeneratorForm: FC = () => {
+  const { twitterHandle, setTwitterHandle, loading, publicWallet, setPublicWallet } =
+    useNFTNameGenerator()
```

Comment:
removed state props and used the new provider instead

Reply:
love it